src/nnetbin/append-post-to-feats.cc

```diff
@@ -0,0 +1,145 @@
+// featbin/append-post-to-feats.cc
+
+// Copyright 2016 Brno University of Technology (Author: Karel Vesely)
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "base/kaldi-common.h"
+#include "util/common-utils.h"
+#include "matrix/kaldi-matrix.h"
+#include "nnet/nnet-utils.h"
+
+namespace kaldi {
+namespace nnet1 {
+
+void AppendPostToFeats(const Matrix<BaseFloat> &in,
+                       const Posterior &post,
+                       const int32 post_dim,
+                       Matrix<BaseFloat> *out) {
+  // Check inputs,
+  if (in.NumRows() != post.size()) {
+    KALDI_WARN << "Mismatch of length!"
+               << " features: " << in.NumRows()
+               << " posteriors: " << post.size();
+  }
+  KALDI_ASSERT(in.NumRows() == post.size());
+  // Convert post to Matrix,
+  Matrix<BaseFloat> post_mat;
+  PosteriorToMatrix(post, post_dim, &post_mat);
+  // Build the output matrix,
+  out->Resize(in.NumRows(), in.NumCols() + post_dim);
+  out->ColRange(0, in.NumCols()).CopyFromMat(in);
+  out->ColRange(in.NumCols(), post_mat.NumCols()).CopyFromMat(post_mat);
+}
+
+}  // namespace nnet1,
+}  // namespace kaldi,
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi::nnet1;
+    using namespace kaldi;
+    using namespace std;
+
+    const char *usage =
+        "Append posteriors to features\n"
+        "\n"
+        "Usage: append-post-to-feats <in-rspecifier1> <in-rspecifier2> <out-wspecifier>\n"
+        " or: append-post-to-feats <in-rxfilename1> <in-rxfilename2> <out-wxfilename>\n"
```

Comment:
Usage message should mention options as [options] and should have an example line showing --post-dim.